contracts/allocators/NFTXAllocator.sol

```diff
@@ -0,0 +1,285 @@
+    INFTXLPStaking internal liquidityStaking;
+
+    // Floor Treasury contract
+    ITreasury internal treasury;
+
+    // Corresponding NFTX token vault data for tokens
+    mapping (address => stakingTokenData) public stakingTokenInfo;
+
+    // Corresponding xTokens for tokens
+    mapping (address => dividendTokenData) public dividendTokenInfo;
+
+
+    /**
+     * @notice initialises the construct with no additional logic.
+     */
+
+    constructor (
+        IFloorAuthority _authority,
+        address _inventoryStaking,
+        address _liquidityStaking,
+        address _treasury
+    ) FloorAccessControlled(_authority) {
+        inventoryStaking = INFTXInventoryStaking(_inventoryStaking);
+        liquidityStaking = INFTXLPStaking(_liquidityStaking);
+
+        treasury = ITreasury(_treasury);
+    }
+
+
+    /**
+     * Deprecated in favour of harvestAll(address _token).
+     */
+
+    function harvest(address _token, uint256 _amount) external override {
+        revert("Method is deprecated in favour of harvestAll(address _token)");
+    }
+
+
+    /**
+     * @notice claims rewards from the vault.
+     */
+
+    function harvestAll(address _token) override external {
+        // We only want to allow harvesting from a specified liquidity pool mapping
+        require(stakingTokenInfo[_token].exists, "Unsupported token");
+        require(stakingTokenInfo[_token].isLiquidityPool, "Must be liquidity staking token");
+
+        // Trigger our rewards to be claimed
+        liquidityStaking.claimRewards(stakingTokenInfo[_token].vaultId);
+
+        // Get the reward token for this stakingToken
+        address _rewardToken = stakingTokenInfo[_token].rewardToken;
+        
+        // Deposit the harvested rewards into the treasury
+        uint256 balance = IERC20(_rewardToken).balanceOf(address(this));
+        uint256 value = treasury.tokenValue(_rewardToken, balance);
+
+        // Approve and deposit asset into treasury
+        IERC20(_rewardToken).approve(address(treasury), balance);
+
+        // Pass the tokenValue as profit to stop the treasury minting FLOOR
+        treasury.deposit(balance, _rewardToken, value);
+    }
+
+
+    /**
+     * @notice sends any ERC20 token in the contract to caller.
+     */
+
+    function rescue(address _token) external override onlyGovernor {
+        // If the token is known, then we shouldn't be able to rescue it
+        require(!stakingTokenInfo[_token].exists, "Known token cannot be rescued");
+
+        // Get the amount of token held on contract
+        uint256 _amount = IERC20(_token).balanceOf(address(this));
+
+        // Confirm that we hold some of the specified token
+        require(_amount > 0, "Token not held in contract");
+
+        // Send to Governor
+        IERC20(_token).safeTransfer(msg.sender, _amount);
+    }
+
+
+    /**
+     * @notice withdraws asset from treasury, deposits asset into lending pool,
+     * then deposits xToken into treasury.
```

Comment:
this step appears to be missing — depositing xToken into treasury

but it might be easier to just let the xToken sit in the allocator to save the extra step of shuffling it from allocator <> treasury